modin/engines/base/io/text/text_file_dispatcher.py

```diff
@@ -144,64 +150,108 @@ def partitioned_file(
             Number of rows of file to read.
         skiprows: array or callable, optional
             Specifies rows to skip.
+        pre_reading: dict, optional
+            Specifies the first data chunk that should be read.
+            pre_reading dict should contain the next fields:
+            rows_number - the number of rows that should be read
+            start - start byte of this chunk
+            end - end byte of this chunk
+        read_rows_counter: int, optional
+            The number of rows that already read.
         quotechar: bytes, default b'"'
             Indicate quote in a file.
         is_quoting: bool, default True
             Whether or not to consider quotes.
+        partition_skiprows: int
+            Number of additional rows that should be skipped in every
+            partition.
 
         Returns
         -------
-        An array, where each element of array is a tuple of two ints:
-        beginning and the end offsets of the current chunk.
+        List of tuples with the next elements:
+            int: partition start read byte
+            int: partition end read byte
+            int, array-like or callable: skiprows object aligned (adopted)
+                to the exact partition
+            int: number of rows that exact partition will contain
         """
+        rows_considered = None
+        pre_reading_rows = 0
+        result = []
+
         if num_partitions is None:
             num_partitions = NPartitions.get()
 
-        result = []
-        file_size = cls.file_size(f)
-
-        if skiprows:
-            outside_quotes, read_rows = cls._read_rows(
-                f,
-                nrows=skiprows,
-                quotechar=quotechar,
-                is_quoting=is_quoting,
+        if pre_reading:
+            pre_reading_rows = pre_reading["rows_number"]
+            result.append(
+                (
+                    pre_reading["start"],
+                    pre_reading["end"],
+                    partition_skiprows,
+                    rows_considered,
+                )
             )
+            num_partitions -= 1
```

Comment:
I have a concern about the location of this logic. Don't you think that it would be better to just insert `pre_reading` at the beginning of `splits` in `csv_dispatcher` instead of adding a new argument to the `partitioned_file`, it would make this function lighter.

Reply:
Yes, i think it will be better, `pre_reading` part is left in the `csv_dispatcher`.